### PR TITLE
[global] Update jest from 24 to 29 version

### PR DESCRIPTION
**Pull request by lsroman**
Updated the Jest and its dependencies. The tests have also been updated so that they pass.

## Description

To switch to React 18, you need to update `testing-library`, to update it you need to update the `Jest`.
The screenshots have changed, but only slightly, since the [dependency](https://github.com/americanexpress/jest-image-snapshot/blob/main/CHANGELOG.md#breaking-changes-2) that compares screenshots has been updated.
I also fixed the warnings and errors in the console when running the tests.
When updating, there was confusion in dependencies, so I specified them more explicitly.

## Motivation and Context

I want to upgrade to react 18

## How has this been tested?

Run tests and hope for the best 🤞

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: semcore/d3-chart/__tests__/index.test.tsx [hunk 1]
```diff
@@ -2525,6 +2527,8 @@ describe('d3 charts visual regression', () => {
       );
     };
 
-    expect(await snapshot(<Component />)).toMatchImageSnapshot();
+    expect(await snapshot(<Component />)).toMatchImageSnapshot({
+      failureThreshold: 0.0001,
```
Comment (phytonmk) [inline semcore/d3-chart/__tests__/index.test.tsx:2531]: Do we really need it?

File: semcore/d3-chart/__tests__/index.test.tsx [hunk 2]
```diff
@@ -2441,7 +2441,9 @@ describe('d3 charts visual regression', () => {
       );
     };
 
-    expect(await snapshot(<Component />)).toMatchImageSnapshot();
+    expect(await snapshot(<Component />)).toMatchImageSnapshot({
+      // failureThreshold: 0.0001,
```
Comment (phytonmk) [inline semcore/d3-chart/__tests__/index.test.tsx:2445]: Extra comment

File: tools/playground/src/index.tsx [hunk 3]
```diff
@@ -2,4 +2,12 @@ import React from 'react';
...
 import { App } from './App';
 
+// eslint-disable-next-line ssr-friendly/no-dom-globals-in-module-scope
 ReactDOM.render(<App />, document.getElementById('root'));
+
+// import React from 'react';
+// import { createRoot } from 'react-dom/client';
+// import { App } from './App';
+//
+// const root = createRoot(document.getElementById('root'));
```
Comment (phytonmk) [inline tools/playground/src/index.tsx:12]: React 18 update leaked here

**Review by phytonmk — APPROVED** [review]

File: semcore/d3-chart/__tests__/index.test.tsx (hunk 1 above)
Comment (lsroman) [reply semcore/d3-chart/__tests__/index.test.tsx:2518]: Yes. Rendering circles in svg is not stable when compared by pixel. I added a clarifying comment

File: semcore/d3-chart/__tests__/index.test.tsx (hunk 2 above)
Comment (lsroman) [reply semcore/d3-chart/__tests__/index.test.tsx]: I deleted

File: tools/playground/src/index.tsx (hunk 3 above)
Comment (lsroman) [reply tools/playground/src/index.tsx:12]: Yes, I did it on purpose. There is also commented code in the render site and snapshots. My goal was just to upgrade to react 18.

**Pull request merged**